libraries/chain/include/eos/chain/types.hpp

```diff
@@ -171,6 +171,7 @@ namespace eos { namespace chain {
       transaction_object_type,
       producer_object_type,
       chain_property_object_type,
+      transaction_history_object_type,
```

Comment:
A comment here noting that this object is defined in the account_history_plugin would be good (like in the lines below). With modern IDEs, it's not as necessary, but it's still helpful to note which object types are declared here, but implemented in a separate library/plugin.